tapir/shifts/templates/shifts/shift_calendar_template.html

```diff
@@ -10,56 +10,79 @@
-                    url += "&date_to=" + document.getElementById("date-to").value;
-                    window.location = url;
-                }
-            </script>
-            <div class="d-flex align-items-baseline gap-2">
-                <label for="date-from" class="form-label">{% translate "From" %}</label>
-                <input id="date-from"
-                       type="date"
-                       value="{{ date_from }}"
-                       class="form-control"
-                       style="max-width: 160px" />
-            </div>
-            <div class="d-flex align-items-baseline gap-2">
-                <label for="date-to" class="form-label">{% translate "To" %}</label>
-                <input id="date-to"
-                       type="date"
-                       value="{{ date_to }}"
-                       class="form-control"
-                       style="max-width: 160px" />
+    <div class="collapse" id="collapseCalendarInfo">
+        <div class="card-group mb-2">
+            <div class="card">
+                <h5 class="card-header">
+                    {% block card_title %}
+                    {% endblock card_title %}
+                </h5>
+                <div class="card-body">
+                    {% block card_helptext %}
+                    {% endblock card_helptext %}
+                </div>
             </div>
-            <button class="{% tapir_button_link %}" onclick="onFilter()">{% translate "Filter" %}</button>
+            {% shift_filters %}
         </div>
     </div>
     {% for week, shifts_by_day in shifts_by_weeks_and_days.items %}
         <div class="card mb-2">
             {% with week_group=week_to_group|dictionary_get:week %}
-                <h5 class="card-header">
-                    KW {{ week|date:"W" }} {% translate 'starting' %} {{ week }},
-                    {% translate 'Week ' %}{{ week_group.name }}
-                </h5>
+                <div class="card-header">
+                    <div class="container-fluid">
+                        <div class="row">
+                            <div class="col-5">
+                                <h5>
+                                    KW {{ week|date:"W" }} {% translate 'starting' %} {{ week }},
+                                    {% translate 'Week ' %}{{ week_group.name }}
+                                    <div>
+                                        <script>
+                                            function onFilter() {
+                                                let url = "{% url "shifts:calendar" %}";
+                                                url += "?date_from=" + document.getElementById("date-from").value;
+                                                url += "&date_to=" + document.getElementById("date-to").value;
+                                                window.location = url;
+                                            }
+                                        </script>
```

Comment:
This redefines the function for every week. Let's put it in `{% block head %}` instead.

Reply:
off-topic: It feels like the only filter not in python. Was there a reason for Typescript? I remember having problems standardizing the date-format

Reply:
@Theophile-Madet no idea, that was done before me ;)

Reply:
The filtering itself is done in python (see `ShiftCalendarView.get_context_data()`). The difference with the other views with filters, like the member list view, is that those other views are table views, so we can use `django_filters.FilterSet`. The calendar view is not a table so linking with FilterSet is more complex. I think it's still possible, but as long as we only filter dates, I think it's simpler to just add the dates as url parameters.